### DISCUSSION TITLE
A query containing dollar-quoted format string causes SQL editor to run other commands unintentionally (Postgres)

### DISCUSSION THREAD

**Discussion by philosaur** [dbeaver/dbeaver#22543] [Q&A] [answered]
### Description

Running a query (Execute SQL query) that contains a dollar-quoted format string causes DBeaver to run that query as well as all subsequent queries in the same editor--as if "Execute SQL script" had been run instead.

My last version of DBeaver was from probably 6 moths ago, and this behavior began when I upgraded to the Christmas Day version.

### DBeaver Version

Community Edition, vers. 23.3.1.202312241705

### Operating System

Windows 10

### Database and driver

Postgres, (PostgreSQL JDBC Driver)

### Steps to reproduce

1. Click "Execute SQL Script" or press F5 (by default) with this SQL:
```
SELECT format($$%1$s$$, 'foo');

SELECT 'bar';

SELECT 'baz1';

SELECT 'baz2';

SELECT 'baz3';
```
2. DBeaver produce five separate output tabs with output from each respective command, instead of running just the first command.  
3. Switching the dollar quotes to single quotes causes it to only run the first query, as expected.

**Comment by E1izabeth** [comment, marked as answer]
You can switch this preference to get the desired behavior
![image](https://github.com/dbeaver/dbeaver/assets/28875055/4665bdca-c773-484e-9f16-e8b3a62a6bde)

**Comment by philosaur** [comment]
I tried changing this setting like you suggested, and it works.  

Thank you.

**Reply by philosaur** [reply]
I strongly disagree that this issue should be considered handled and/or closed.  [E1izabeth](https://github.com/E1izabeth)'s answer is a temporary stop-gap, at best, and closing this implies that this is expected behavior.  (My original reply here said as much, but then I thought I must be misunderstanding, so edited it into a simple thank you.)

"Show $$ quotes as string/Code block" doesn't even begin to capture what that toggle actually does.  

The label reads like it merely changes rendering of SQL.  

What it actually does is, if you choose "Code block" and then run a dollar-quoted string ahead of other queries in the same window is to _emulate the Execute SQL Script button_ . That's _vastly_ different from what the label says it does.  

Previous versions did not behave this way.  Please, restore the previous behavior, disable the option, or at least change the label to reflect the surprising new behavior.

**Reply by philosaur** [reply]
I don't mean to be argumentative, rude, or demanding, but I'm very new to github and I don't know what it means for this to be moved to discussion.  The fact that the original issue was closed is also confusing.

I love DBeaver, and have had to fight to be allowed to use it for work.  And right now I can't upgrade past 23.3 without knowing I can use dollar-quoted strings like before.

**Comment by E1izabeth** [comment]
Here is the issue about it #22652
